examples/leader-election/test.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+cleanup() {
+  exit_code=$?
+  set +e
+  echo "Rollback metacontroller..."
+  kubectl scale statefulset --replicas="$previous_replicas" -n metacontroller metacontroller
+  kubectl rollout undo statefulset metacontroller -n metacontroller
+  kubectl rollout status --watch --timeout=180s statefulset/metacontroller -n metacontroller
+  exit $exit_code
+}
+trap cleanup EXIT
+
+set -ex
+
+success_msg='successfully acquired lease'
+attempt_msg='attempting to acquire leader lease'
+
+previous_replicas=$(kubectl get statefulset metacontroller -n metacontroller -o=jsonpath='{.spec.replicas}')
+kubectl apply -k ./manifest
+kubectl rollout status --watch --timeout=180s statefulset/metacontroller -n metacontroller
+
+# both pods must be ready before checking logs
+kubectl wait --timeout=180s --for=condition=ready pod metacontroller-0 -n metacontroller
+kubectl wait --timeout=180s --for=condition=ready pod metacontroller-1 -n metacontroller
+
+maximum_attempts=600
+# wait for one pod to acquire the leader lease
+until [[ "$(kubectl logs metacontroller-0 -n metacontroller | grep "$success_msg" | wc -l)" -eq 1 ||
+         "$(kubectl logs metacontroller-1 -n metacontroller | grep "$success_msg" | wc -l)" -eq 1 ]]; do
+  sleep 5
+  # timeout at 5m if no leader lease acquired
```

Comment:
Is this correct? 
i.e. 600*5 seconds ~ 50 minutes

Reply:
You are right, I goofed up that one  😄 . I changed this to 36 and 5 to equal 180s timeout similar to other timeouts in the test.